http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/AutopauseMonitor.scala

```diff
@@ -0,0 +1,84 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package monitor
+
+import cats.effect.{Async, ContextShift, Timer}
+import cats.implicits._
+import fs2.Stream
+import fs2.concurrent.InspectableQueue
+import io.chrisdavenport.log4cats.Logger
+import org.broadinstitute.dsde.workbench.leonardo.config.AutoFreezeConfig
+import org.broadinstitute.dsde.workbench.leonardo.dao.JupyterDAO
+import org.broadinstitute.dsde.workbench.leonardo.db._
+import org.broadinstitute.dsde.workbench.leonardo.http.{dbioToIO, nowInstant}
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
+
+import scala.concurrent.ExecutionContext
+
+/**
+ * This monitor periodically sweeps the Leo database and checks for clusters which no longer exist in Google.
+ */
+class AutopauseMonitor[F[_]: ContextShift: Timer](
+  config: AutoFreezeConfig,
+  jupyterDAO: JupyterDAO[F],
+  publisherQueue: InspectableQueue[F, LeoPubsubMessage]
+)(implicit F: Async[F],
+  metrics: OpenTelemetryMetrics[F],
+  logger: Logger[F],
+  dbRef: DbReference[F],
+  ec: ExecutionContext) {
+
+  val process: Stream[F, Unit] =
+    (Stream.sleep[F](config.autoFreezeCheckInterval) ++ Stream.eval(autoPauseCheck)).repeat
+
+  private[monitor] val autoPauseCheck: F[Unit] =
```

Comment:
yea, this is mostly straight copy out of ClusterMonitorSupervisor class. Those unit tests for autopause were living in `ClusterMonitorSupervisorSpec` as well, which is now deleted since all it had is autopause unit tests